### PR TITLE
Game state equality

### PR DESCRIPTION
**Pull request by invisime**
This sets up up to handle state comparisons. See commit messages for more details.

### CODE REVIEW

File: GameEngineTests/GameBoardTests.cs [hunk 1]
```diff
@@ -6,6 +6,40 @@ namespace GameEngineTests
...
+            var initialBoard = new GameBoard(6);
+            var clonedBoard = initialBoard.Clone();
+
+            clonedBoard.Board.Remove(0);
+
+            Assert.AreNotEqual(initialBoard, clonedBoard);
+        }
+
+        [TestMethod]
+        public void ShouldNotBeEqualIfTheParlimentsAreNotEqual()
```
Comment (rpherbig) [inline GameEngineTests/GameBoardTests.cs]: 💡 `...Parliaments...`

**Review by rpherbig — COMMENTED** [review]
💡 I agree that DRYing this up would be useful. Not necessary for this PR IMO.

❓ Where did the GetHashCode magic numbers come from? (I don't doubt that they're ok as-is, I just want to become more familiar with overriding GetHashCode)

**Comment by invisime** [comment]
> ❓ Where did the GetHashCode magic numbers come from? (I don't doubt that they're ok as-is, I just want to become more familiar with overriding GetHashCode)

I used [Visual Studio's generator](https://docs.microsoft.com/en-us/visualstudio/ide/reference/generate-equals-gethashcode-methods?view=vs-2019) and just trusted that the numbers were suitable.

**Comment by rpherbig** [comment]
Neat, I wasn't aware VS could do that. Thanks!

**Pull request merged**